### PR TITLE
Inline markdown elements: emphasis, strong, link

### LINKED ISSUE #778: More markdown support?
I'm trying to understand why bikeshed supports [a subset of markdown](https://github.com/tabatkins/bikeshed/blob/master/docs/markup.md) and not other parts. I'm not sure I get a trend beyond what's currently implemented is what's there.

 Specifically, I'd like to support:
* Regular markdown links à la `[awesome](zombo.com "Anything is possible")` as well as with ID.
* Emphasis and bold with `*` and `**`.

Is that just something I can send a PR for, or do you have a moral objection?

### PR DESCRIPTION
**Pull request by jfbastien**
Fixes #778.

### CODE REVIEW

File: tests/markdown011.html [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  <div data-fill-with="at-risk"></div>
+  <nav data-fill-with="table-of-contents" id="toc">
+   <h2 class="no-num no-toc no-ref" id="contents">Table of Contents</h2>
+  </nav>
+  <main>
+   <p>Text is <em>wonderful</em>. <strong>Amazing</strong> even.</p>
+   <p>Text is <em>wonderful</em>. <strong>Amazing</strong> even.</p>
+   <p><strong>SO GREAT</strong></p>
+   <p>un<em>frigging</em>believable</p>
+   <p>You can \*escape\* things. Except <strong>death</strong> and <em>taxes</em>.</p>
```
Comment (mkruisselbrink) [inline tests/markdown011.html]: It isn't really escaping things if that leaves the escape character in the resulting output...

File: bikeshed/shorthands.py [hunk 2]
```diff
@@ -226,6 +226,16 @@ def elementReplacer(match):
     def varReplacer(match):
         return E.var(match.group(1))
 
+    strongRe = re.compile(r"([_\*])\1([\d\w\s]+)\1\1")
```
Comment (tabatkins) [inline bikeshed/shorthands.py]: No need to escape a `*` inside of a character class.

`\d\w\s` does not capture the full range of things you can put in there - for example, it excludes hyphens. You're gonna want a lookahead that just checks that the next two characters aren't `\1\1`.

File: bikeshed/shorthands.py [hunk 3]
```diff
@@ -226,6 +226,16 @@ def elementReplacer(match):
...
+
+    def strongReplacer(match):
+        return E.strong(match.group(2))
+
+    emRe = re.compile(r"([_\*])([\d\w\s]+)\1")
```
Comment (tabatkins) [inline bikeshed/shorthands.py]: Same as above.

**Comment by jfbastien** [comment]
I addressed a bunch of the comments, and followed some of commonmark:
http://spec.commonmark.org/0.26/#emphasis-and-strong-emphasis

I ignored the punctuation character part of delimiter runs, and can't handle nesting because that's a limitation in the current bikeshed code.

File: tests/markdown011.html (hunk 1 above)
Comment (jfbastien) [inline tests/markdown011.html]: Done.

File: bikeshed/shorthands.py (hunk 2 above)
Comment (jfbastien) [inline bikeshed/shorthands.py]: Done.

File: bikeshed/shorthands.py (hunk 3 above)
Comment (jfbastien) [inline bikeshed/shorthands.py]: Done.

**Comment by jfbastien** [comment]
I added support for inline links as well. Again, not full markdown but close.

**Pull request merged**